python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -373,18 +383,23 @@ def _getItem(self, index):
                 except Exception as err:
 
                     try:
+                        logger.warning("Loading subjob #%s from backup, recent changes may be lost" % index)
                         subjob_data = self.__get_dataFile(str(index), True)
+                        sj_file = self._loadSubJobFromDisk(subjob_data)
                         loaded_sj = from_file(sj_file)[0]
+                        has_loaded_backup = True
                     except Exception as err:
                         logger.debug("Failed to Load XML for job: %s using: %s" % (str(index), str(subjob_data)))
                         logger.debug("Err:\n%s" % str(err))
                         raise
 
-                # if load was successful then set parent and add to the _cachedJobs dict
-                if loaded_sj:
-                    loaded_sj._setParent( self._definedParent )
+                loaded_sj._setParent( self._definedParent )
+                if has_loaded_backup:
+                    loaded_sj._setDirty()
+                    loaded_sj._getParent()._setDirty()
```

Comment:
Typo, was left in from dealing with the job.master attribute which made this step look needed.